### PR TITLE
Comma as separator in calculator

### PR DESCRIPTION
**Pull request by Lukentui**
#49 
Maybe u talk about another problems, but I don't saw anything in the implementation.
https://imgur.com/a/ah6eoiQ

### CODE REVIEW

**Comment by oliverschwendener** [comment]
What happens if you do this: `pow(2,6)`?

image

**Comment by oliverschwendener** [comment]
By the way you can add images directly in the GitHub comment ;)

**Comment by Lukentui** [comment]
> 
> 
> By the way you can add images directly in the GitHub comment ;)

It's really conveniently :D

File: src/main/plugins/calculator-plugin/calculator-plugin.ts [hunk 1]
```diff
@@ -56,6 +56,13 @@ export class CalculatorPlugin implements ExecutionPlugin {
...
 
     private getInputNumber(userInput: string): string {
-        return userInput.replace(",", ".");
+        const inputWithoutCommas: string = userInput.replace(/,/g, ".");
+
+        if (inputWithoutCommas.indexOf("(") === -1 && inputWithoutCommas.indexOf(")") === -1) {
+            return inputWithoutCommas;
+        } else {
+            // replace all `.` to `,` in brackets
+            return inputWithoutCommas.replace(/(\(.*?)\.(.*?\))/g, "$1,$2");
```
Comment (Lukentui) [inline src/main/plugins/calculator-plugin/calculator-plugin.ts]: This solves the problem, but needs testings.

test

**Comment by Lukentui** [comment]
This solves the problem, but needs testings.
img

**Comment by oliverschwendener** [comment]
The calculator plugin heavily depends on [mathjs](https://mathjs.org/) and if we start tweaking the input that the user provides we'll maybe break it's functionality. As far as I know mathjs uses a `,` for separating values in an array evaluation and `.` for the decimal separator.

This is why I'm not willing to merge this PR just for the sake of being able to write `,` instead of `.`.

**Comment by MetaMmodern** [comment]
> What happens if you do this: `pow(2,6)`?
> 
> image

Wow. Hi Oliver and thank you for your great application, I use it as a fast calculator every day))). I was looking for another issue actually. I'd like ueli to automatically detect comma or period delimeter for numbers. I mean not the delimeter for thousands, but for the part of the number which is less than one(like for you it is after period: .23, for me it is after comma: ,23)

But now I see why it is almost impossible, you've implemented functions inside ueli, so commas are parsed as delimeters for expressions, not for numbers. But I have a question anyway. Is there any way you could do this? It is very frustrating to change language every time I open and close ueli. I write mostly in Russian or Ukrainian, but when I open ueli the standard delimeter(comma for me) does not work(

Yes, I know that I can change the delimeter in settings, but consider the opposite situation, when I type smth in a doc in English, open ueli and start fast calculation with a comma, it will be the same, I'd have to change the language again((

(Also, I use numpad when calucaltion something, so the comma button on it changes to point when a switch to English(telling this just in case if it matters))

**Comment by tkohlmeier** [comment]
When the decimal separator is set to comma, the argument separator is semicolon.

So this should work: `pow(2;6)`

**Comment by MetaMmodern** [comment]
> When the decimal separator is set to comma, the argument separator is semicolon.
> 
> So this should work: `pow(2;6)`

Yes, I understand, but consider the opposite situation. I use 2 languages: Russian and English and the seperator on numpad corresponds to different symbols on different languages. In Russian it is comma, but in English it is period. Maybe you have some ideas to use as a workaround for this?

**Comment by MetaMmodern** [comment]
> Alright, I misunderstood you.
> 
> I think ueli should just use the system decimal separator. So instead of having to manually adjust it, it should detect that the input language has changed and change the decimal separator accordingly. That should fit you needs, right?

Yeeeees, totally, right in the spot! That would be great. 
Wow, you've described the solution so clearly, amazing))) 
Are you a contributor to this repo?

**Comment by tkohlmeier** [comment]
Yes, among other things I made the decimal separator selectable (#541).

When I find some time I will try to implement this.

**Comment by MetaMmodern** [comment]
> Yes, among other things I made the decimal separator selectable (#541).
> 
> When I find some time I will try to implement this.

Awesome, mention me in PR then please, I'll request some donation addresses 😄🤡
I write in ts too, but I have no experience in electron and vue, otherwise I'd try to implement it in a fork 😅

**Pull request closed without merging**